nautobot/extras/api/views.py

```diff
@@ -504,69 +564,23 @@ class JobRunTokenPermissions(TokenPermissions):
-    """Deprecated Job-class-based views. Use Job-model-based views instead."""
-
-    permission_classes = [IsAuthenticated]
-    lookup_field = "class_path"
-    lookup_value_regex = "[^/]+/[^/]+/[^/]+"  # e.g. "git.repo_name/module_name/JobName"
-
-    def list(self, request):
-        if not request.user.has_perm("extras.view_job"):
-            raise PermissionDenied("This user does not have permission to view jobs.")
-        job_content_type = ContentType.objects.get(app_label="extras", model="job")
-        results = {
-            r.name: r
-            for r in JobResult.objects.filter(
-                obj_type=job_content_type,
-                status__in=JobResultStatusChoices.TERMINAL_STATE_CHOICES,
-            )
-            .defer("data")
-            .order_by("created")
-        }
-
-        job_models = Job.objects.restrict(request.user, "view")
-        jobs_list = [
-            job_model.job_class()  # TODO: why do we need to instantiate the job_class?
-            for job_model in job_models
-            if job_model.installed and job_model.job_class is not None
-        ]
-        for job_instance in jobs_list:
-            job_instance.result = results.get(job_instance.class_path, None)
-
-        serializer = serializers.JobClassSerializer(jobs_list, many=True, context={"request": request})
-
-        return Response(serializer.data)
-
-    def retrieve(self, request, class_path):
-        if not request.user.has_perm("extras.view_job"):
-            raise PermissionDenied("This user does not have permission to view jobs.")
-        try:
-            job_model = Job.objects.restrict(request.user, "view").get_for_class_path(class_path)
-        except Job.DoesNotExist:
-            raise Http404
-        if not job_model.installed or job_model.job_class is None:
-            raise Http404
-        job_content_type = ContentType.objects.get(app_label="extras", model="job")
-        job = job_model.job_class()  # TODO: why do we need to instantiate the job_class?
-        job.result = JobResult.objects.filter(
-            obj_type=job_content_type,
-            name=job.class_path,
-            status__in=JobResultStatusChoices.TERMINAL_STATE_CHOICES,
-        ).first()
-
-        serializer = serializers.JobClassDetailSerializer(job, context={"request": request})
-
-        return Response(serializer.data)
-
-    @swagger_auto_schema(method="post", request_body=serializers.JobInputSerializer)
-    @action(detail=True, methods=["post"])
-    def run(self, request, class_path):
+    @swagger_auto_schema(
+        responses={"200": serializers.JobClassDetailSerializer()}, operation_id="extras_jobs_run_deprecated"
+    )
+    @action(
+        detail=False,  # a /jobs/... URL, not a /jobs/<pk>/... URL
+        methods=["post"],
+        permission_classes=[JobRunTokenPermissions],
+        url_path="(?P<class_path>[^/]+/[^/]+/[^/]+)/run",
+        url_name="run",
+    )
+    def run_deprecated(self, request, class_path):
```

Comment:
Same here, how is `run_deprecated` being called anywhere? I don't see any references to it!